src/lib/ptree/json.lua

```diff
@@ -86,7 +108,51 @@ local function read_json_string(input)
    return table.concat(parts)
 end
 
-function read_json_object(input)
+read_json_array = function(input)
+   skip_whitespace(input)
+   consume(input, "[")
+   skip_whitespace(input)
+   local ret = {}
+   if not check(input, "]") then
+      repeat
+         skip_whitespace(input)
+         local v = read_json_type(input)
+         skip_whitespace(input)
+         ret[#ret+1] = v
```

Comment:
Same as `table.insert(ret, v)`. The advantage is that it avoids computing #ret per iteration.